src/databricks/labs/blueprint/installation.py

```diff
@@ -481,7 +482,11 @@ def _marshal(self, type_ref: type, path: list[str], inst: Any) -> tuple[Any, boo
         if dataclasses.is_dataclass(type_ref):
             return self._marshal_dataclass(type_ref, path, inst)
         if type_ref == list:
-            return self._marshal_list(type_ref, path, inst)
+            return self._marshal_raw_list(path, inst)
+        if type_ref == dict:
+            return self._marshal_raw_dict(path, inst)
```

Comment:
I think this change means that the generic type parameter for lists and dictionaries (eg. `list[str]` or `dict[str, str]`) is now ignored: previously this was used for marshalling, but now we just treat these as "raw" lists, is that correct?

Reply:
No it doesn't mean that. It actually fixes a bug where an incorrect `type_ref` was passed down.

Reply:
That's fixed now